test/gurka/test_graphfilter.cc

```diff
@@ -0,0 +1,601 @@
+    const DirectedEdge* OD_edge = nullptr;
+    std::tie(DO_edge_id, DO_edge, OD_edge_id, OD_edge) =
+        findEdge(graph_reader, map.nodes, "DO", "O", baldr::GraphId{});
+    EXPECT_NE(DO_edge, nullptr);
+    EXPECT_NE(OD_edge, nullptr);
+  }
+
+  // more edges will exist
+  result = gurka::do_action(valhalla::Options::route, map, {"A", "L"}, "auto");
+  gurka::assert::raw::expect_path(result, {"East Fort Avenue", "East Fort Avenue", "East Fort Avenue",
+                                           "East Fort Avenue", "East Fort Avenue", "East Fort Avenue",
+                                           "East Fort Avenue", "East Fort Avenue", "East Fort Avenue",
+                                           "East Fort Avenue", "East Fort Avenue"});
+}
+
+TEST(Standalone, SimpleFilter2) {
+
+  constexpr double gridsize_metres = 10;
+
+  const std::string ascii_map = R"(
+
+        J----K----L----M----------------------N
+        |    |    |    |                      |
+        |    H    I    |                      |
+        |    |    |    |                      |
+        A----B----C----D----E------------F----G
+        |              |     \          /     |
+        |              |      \        /      |
+        O              P       Q------R       S                  
+  )";
+
+  const gurka::ways ways = {
+      {"JKLMN",
+       {{"highway", "secondary"},
+        {"osm_id", "100"},
+        {"bicycle", "no"},
+        {"foot", "no"},
+        {"name", "Vondellaan"},
+        {"oneway", "yes"}}},
+      {"ABCDEFG", {{"highway", "cycleway"}, {"osm_id", "101"}, {"oneway", "yes"}}},
+      {"JAO",
+       {{"highway", "residential"},
+        {"osm_id", "102"},
+        {"bicycle", "no"},
+        {"foot", "no"},
+        {"name", "Croesestraat"},
+        {"oneway", "yes"},
+        {"oneway:bicycle", "no"}}},
+      {"BHK", {{"highway", "cycleway"}, {"osm_id", "103"}, {"oneway", "yes"}}},
+      {"CIL", {{"highway", "footway"}, {"osm_id", "104"}}},
+      {"MDP", {{"highway", "service"}, {"osm_id", "105"}, {"access", "private"}}},
+      {"EQRF",
+       {{"highway", "service"}, {"osm_id", "106"}, {"access", "delivery"}, {"oneway", "yes"}}},
+      {"NG", {{"highway", "unclassified"}, {"osm_id", "107"}, {"name", "Oosterkade"}}},
+      {"GS", {{"highway", "cycleway"}, {"osm_id", "108"}}},
+  };
+
+  const gurka::nodes nodes = {
+      {"H", {{"highway", "traffic_signals"}}},
+      {"I", {{"highway", "traffic_signals"}, {"traffic_signals:direction", "backward"}}},
+  };
+
+  auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize_metres, {5.11668, 52.07826});
+  auto map =
+      gurka::buildtiles(layout, ways, nodes, {}, work_dir,
+                        {{"mjolnir.admin", {VALHALLA_SOURCE_DIR "test/data/language_admin.sqlite"}},
+                         {"mjolnir.include_pedestrian", "false"}});
+  // CIL should be deleted
+  {
+    GraphReader graph_reader = GraphReader(map.config.get_child("mjolnir"));
```

Comment:
nit: would suffice to have one graph reader for the whole test

Reply:
This is not the case in all places as we are reconverting the data multiple times.  So I can reuse when we can do so.

Reply:
jep all good, doesn't really hurt with the mini maps